store/root.js

```diff
@@ -100,54 +107,51 @@ class Root extends Store {
     this.organisation.retrieve()
   }
 
-  @action async changeDataProvider(id) {
+  @action changeDataProvider(id) {
     // Clean-up or initial request
     if (id == null) {
       this.dataProvider = null
       return
     }
 
-    const dataProviderInt = parseInt(id, 10)
-
     // Probably a repeated request. No need to change
-    // Compare strings and numbers
-    // since we do not really know what type of ID the API exposes
-    if (this.dataProvider?.id === dataProviderInt) return
-
-    let dataProviderMetaData = this.user.getDataProviderById(dataProviderInt)
+    //
+    // Compare strings and numbers since we do not know what type of ID
+    // the API exposes
+    // eslint-disable-next-line eqeqeq
+    if (this.dataProvider?.id == id) return
 
     // Check access rights
-    if (
-      dataProviderMetaData !== null &&
-      !this.user.canManage(dataProviderMetaData?.id)
-    ) {
-      if (!this.user.superAdmin) {
-        const dpStr = `DataProvider#${id}`
-        throw new AccessError(
-          `${this.user} does not have access to the ${dpStr}`
-        )
-      }
-
-      try {
-        const { data } = await this.user.retrieveDataProviderMetadata(id)
-        dataProviderMetaData = data
-        this.user.dataProviders = [
-          ...this.user.dataProviders,
-          dataProviderMetaData,
-        ]
-      } catch (error) {
-        throw new AccessError(
-          `Failed to load DataProvider#${id} metadata. Server responded with ${error.status}`
-        )
-      }
+    if (!this.user.canManage(id)) {
+      const dpStr = `DataProvider#${id}`
+      throw new AccessError(`${this.user} does not have access to the ${dpStr}`)
     }
 
-    this.dataProvider = new DataProvider(dataProviderMetaData, {
+    const dataProviderInit = this.findDataProvider(id)
+    this.dataProvider = new DataProvider(dataProviderInit, {
       ...this.options,
       prefetch: true,
     })
   }
 
+  findDataProvider(id) {
+    const dataProvider = this.user.dataProviders.find(
+      // Comparing string ID from the URL with whatever it is in the object
+      // eslint-disable-next-line eqeqeq
+      ({ id: dataProviderId }) => dataProviderId == id
+    )
+    return dataProvider ?? { url: `/data-providers/${id}` }
```

Comment:
you need to fetch new repo metadata hand append them to all data providers array, if it is work in progress sorry abou my review.

Reply:
Why do you think so?

Reply:
because the repository select works like that. 

or you will need to play with observables a little bit. Something like this (I haven't tested it so I may be talking  a bulshit 😄 )


```
class DataProvider extends Resource {
  @observable id = -1

  @observable name = 'Unknown repository'
```

Reply:
Yes, this solves the problem with observing the value. Thank you